src/base/name/chain.rs

```diff
@@ -200,9 +215,27 @@ where
 
 //--- Display
 
-impl<L: fmt::Display, R: fmt::Display> fmt::Display for Chain<L, R> {
+impl<L, R> fmt::Display for Chain<L, R>
+where
+    Self: ToLabelIter,
+{
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}.{}", self.left, self.right)
+        let mut empty = true;
+        for label in self.iter_labels() {
+            if label.is_root() {
```

Comment:
Only the last label should ever be a root, is that correct? Yet this is checked for every label, does that make sense?

Reply:
I can’t think of a way to only check on the last label. As in: how do I know a label is last? (This check is extremely cheap, basically checking if the label is empty.)

Reply:
Maybe using [`split_last()`](https://doc.rust-lang.org/std/primitive.slice.html#method.split_last) ?

Reply:
That and it also iterates over the whole thing to find the last label.